### ISSUE TITLE
Question: should this be used if I wish to show animated GIF or video file in live wallpaper?

### ISSUE THREAD

**Issue by AndroidDeveloperLB** [GLWallpaperService/GLWallpaperService#14]
Or is it possible to do it with a normal one?
If it's possible only by using GLWallpaperService, may I ask how ?
I've searched in so many places, and I don't see a single place to show how to show video files content on live wallpaper.

**Comment by markfguerra** [comment]
@AndroidDeveloperLB I haven't yet looked into showing GIFs/videos in a wallpaper either, but I can say that GLWallpaperService isn't the tool I'd personally use for the job. Although I guess in theory it could be done, this tool is more about sending OpenGL API calls, like rendering graphics as an OpenGL video game would.

**Comment by markfguerra** [comment]
@AndroidDeveloperLB For posterity, did you find anything useful when you looked into this?

**Comment by AndroidDeveloperLB** [comment]
Thank you.
I got how to do it without OpenGL at all. What I've found was a bit buggy (especially the one for showing a video), but in the end I got it to work. Now my issue with it is how to scale it in various ways, because it's stretched:
video:
https://stackoverflow.com/q/50091878/878126
GIF:
https://stackoverflow.com/q/49944661/878126

I wonder if I can find a way to show animated HEIF and WEBP, which are less common these days.
